src/docker-client.ts

```diff
@@ -54,6 +56,7 @@ export class DockerodeClient implements DockerClient {
   public async create(options: CreateOptions): Promise<Container> {
     log.info(`Creating container for image: ${options.repoTag}`);
 
+    const containerNameOption = options.name !== undefined ? { name: options.name } : {};
```

Comment:
The `name` field in Dockerode's `ContainerCreateOptions` is also optional, so we don't need to transform it. 
This should work:

```typescript
const dockerodeContainer = await this.dockerode.createContainer({
      name: options.name,
      Image: options.repoTag.toString(),
      ...
```